plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/data/LayoutOptionData.java

```diff
@@ -99,8 +101,11 @@
     /**
      * Create a layout option data entry.
      */
-    public LayoutOptionData(final String aid, final String aname, final String adescription, final Object adefaultValue,
-            final Class<?> atype, final Set<Target> atargets, final Visibility avisibility) {
+    // SUPPRESS CHECKSTYLE NEXT ParameterNumber
+    public LayoutOptionData(final String aid,
+            final String aname, final String adescription, final Object adefaultValue,
+            final Class<?> atype, final Set<Target> atargets, 
+            final Visibility avisibility, final String[] alegacyIds) {
```

Comment:
If `alegacyIds` was a variable-length argument, clients wouldn't be required to always pass `null`, which would be a good idea since this feature will be pretty much deprecated from the start.